tests_e2e/tests/scripts/ext_cgroups-check_cgroups_extensions.py

```diff
@@ -0,0 +1,204 @@
+        # 2021-11-14T13:09:59.351961Z INFO ExtHandler ExtHandler Started tracking cgroup Microsoft.Azure.Extensions.Edp.GATestExtGo-1.0.0.2
+        # [/sys/fs/cgroup/cpu,cpuacct/azure.slice/azure-vmextensions.slice/azure-vmextensions-Microsoft.Azure.Extensions.Edp.GATestExtGo_1.0.0.2.slice]
+        cgroup_tracked_match = CGROUP_TRACKED_PATTERN.findall(record.message)
+        if len(cgroup_tracked_match) != 0:
+            name, path = cgroup_tracked_match[0][0], cgroup_tracked_match[0][1]
+            if name.startswith(GATESTEXT_FULL_NAME):
+                gatestext_cgroups_tracked = True
+            elif name.startswith(AZUREMONITOREXT_FULL_NAME):
+                azuremonitoragent_cgroups_tracked = True
+            elif name.startswith(GATESTEXT_SERVICE):
+                gatestext_service_cgroups_tracked = True
+            elif name.startswith(AZUREMONITORAGENT_SERVICE) or name.startswith(MDSD_SERVICE):
+                azuremonitoragent_service_cgroups_tracked = True
+            cgroups_added_for_telemetry.append((name, path))
+
+    # agent, gatest extension, azuremonitor extension and extension service cgroups
+    if len(cgroups_added_for_telemetry) < 1:
+        raise Exception('Expected cgroups were not tracked, according to the agent log. '
+                        'Pattern searched for: {0}\n{1}'.format(CGROUP_TRACKED_PATTERN.pattern, cgroups_added_for_telemetry))
+
+    if not gatestext_cgroups_tracked:
+        raise Exception('Expected gatestext cgroups were not tracked, according to the agent log. '
+                        'Pattern searched for: {0}\n{1}'.format(CGROUP_TRACKED_PATTERN.pattern, cgroups_added_for_telemetry))
+
+    if not azuremonitoragent_cgroups_tracked:
+        raise Exception('Expected azuremonitoragent cgroups were not tracked, according to the agent log. '
+                        'Pattern searched for: {0}\n{1}'.format(CGROUP_TRACKED_PATTERN.pattern, cgroups_added_for_telemetry))
+
+    if not gatestext_service_cgroups_tracked:
+        raise Exception('Expected gatestext service cgroups were not tracked, according to the agent log. '
+                        'Pattern searched for: {0}\n{1}'.format(CGROUP_TRACKED_PATTERN.pattern, cgroups_added_for_telemetry))
+
+    if not azuremonitoragent_service_cgroups_tracked:
+        raise Exception('Expected azuremonitoragent service cgroups were not tracked, according to the agent log. '
+                        'Pattern searched for: {0}\n{1}'.format(CGROUP_TRACKED_PATTERN.pattern, cgroups_added_for_telemetry))
+    logging.info("\tExtension cgroups tracked as expected\n%s", cgroups_added_for_telemetry)
+
+
+try:
+    logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.DEBUG, stream=sys.stdout)
+    exit_if_cgroups_not_supported()
+    print_processes()
+
+    verify_agent_cgroup_assigned_correctly()
+    check_temporary_folder()
+    verify_custom_script_cgroup_mount()
+    verify_agent_cgroup_assigned_correctly_after_ama()
```

Comment:
I see this  logs the processes in '/var/lib/waagent/tmp/ps_check_after_ama' and then runs verify_agent_cgroup_assigned_correctly.

In what case would this produce different results than the call to verify_agent_cgroup_assigned_correctly on line 187